python/src/nanoarrow/_array.pyx

```diff
@@ -249,6 +294,24 @@ cdef class CArrayView:
 
         return dictionary
 
+    def _iter_bytes(self, int64_t offset, int64_t length):
```

Comment:
```suggestion
    def _iter_bytes(self, int64_t offset, int64_t length) -> bytes | None:
```

There _might_ be some performance benefit from Cython to declaring the return type here, especially since almost everything else looks to be typed.